### PR TITLE
Feature list dialog

### PR DESCRIPTION
**Pull request by ali-moussavi**
### What this PR does (required):
- added the list dialog
- added the ability to add products using the list dialog
- added the ability to remove the products from each list 
- changed the Product model to have a salePrice attribute and a priceHistory attribute that will be an array
- changed the name of the price attribute in Product model to originalPrice
- changed the addProduct backend route to work with the new Product model attributes

### Screenshots / Videos (required):

https://user-images.githubusercontent.com/48439577/120586455-93e1eb00-c401-11eb-9630-f3edaabcc04a.mp4

### Any information needed to test this feature (required):
- sign in and try adding and deleting products from each list

### Any issues with the current functionality (optional):
The confirm button doesn't do anything right now and the product is added before the user presses confirm but ideally, the product should be added after the user confirms but that would require additional routes on the backend

### CODE REVIEW

File: client/src/components/ListDetailDialog/ListDetailDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+  const [showAddNewItem, setShowAddNewItem] = useState(false);
+  const [listState, setListState] = useState<ListInterface>(list);
+  const [newItemUrl, setNewItemUrl] = useState('');
+  const [urlError, setUrlError] = useState<string>('');
+  const [selectedList, setSelectedList] = useState(list._id);
+  const [newProductImageUrl, setNewProductImageUrl] = useState(
+    'https://images-na.ssl-images-amazon.com/images/I/51RRK1PHUOL._AC_SX425_.jpg',
+  );
+  const [newProductName, setNewProductName] = useState('Microsoft Sculpt Ergonomic Keyboard for Business (5KV-00001)');
+  const [newProdutPrice, setNewProdutPrice] = useState('$79.83');
```
Comment (dproc96) [inline client/src/components/ListDetailDialog/ListDetailDialog.tsx]: Should these defaults be empty strings?

File: client/src/components/ListDetailDialog/ListDetailDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+                  <ArrowBackIcon fontSize="large" />
+                </IconButton>
+                <Button
+                  endIcon={submitting ? <CircularProgress className={classes.buttonSpinner} /> : undefined}
+                  onClick={handleSubmit}
+                  color="primary"
+                  variant="contained"
+                  className={classes.createButton}
+                >
+                  CREATE LIST
```
Comment (dproc96) [inline client/src/components/ListDetailDialog/ListDetailDialog.tsx]:
```suggestion
                  CREATE PRODUCT
```

File: server/controllers/auth.js [hunk 3]
```diff
@@ -7,124 +7,124 @@ const List = require("../models/List");
...
-          username: user.username,
-          email: user.email
-        }
-      }
-    });
-  } else {
-    res.status(401);
-    throw new Error("Invalid email or password");
-  }
+	const { email, password } = req.body;
```
Comment (dproc96) [inline server/controllers/auth.js]: I think your prettier settings aren't aligned with the project settings, can you check that?

**Review by dproc96 — COMMENTED** [review]
Overall this looks really good, had a couple of notes below. Also just a general note for the future, the scope of this PR is really big, it probably could/should have been 2 or 3 PRs. Fine to leave as is for now, but going forward think about the best way to separate the logic of a feature into smaller chunks

**Comment by ali-moussavi** [comment]
> Overall this looks really good, had a couple of notes below. Also just a general note for the future, the scope of this PR is really big, it probably could/should have been 2 or 3 PRs. Fine to leave as is for now, but going forward think about the best way to separate the logic of a feature into smaller chunks

Yes, you are absolutely right. I will keep that in mind for future PRs.

File: server/controllers/auth.js (hunk 3 above)
Comment (ali-moussavi) [reply server/controllers/auth.js]: I'll check

File: client/src/components/ListDetailDialog/ListDetailDialog.tsx (hunk 1 above)
Comment (ali-moussavi) [reply client/src/components/ListDetailDialog/ListDetailDialog.tsx]: yep, you're right. I forgot to delete these after testing

**Review by dproc96 — APPROVED** [review]
Screen Shot 2021-06-08 at 8 25 05 AM
Looks good to me, I'm getting a weird error when I try to scrape though. Don't think it's related but might be something for you guys to look into next

**Pull request merged**